src/button.ts

```diff
@@ -0,0 +1,18 @@
+import { Button } from '@bigtest/interactor';
+
+export default Button.extend('button')
+  .filters({
+    iconLabel: (el) => el.querySelector(".MuiButton-label")?.firstElementChild?.classList.value,
+    customRippleClass: (el) => {
```

Comment:
@cowboyd I added this filter. Prior to this I was using a Span Interactor that was created inline in the test file to do something like this.

```js 
button.find(Span({ className: including(...) }))

Which way do you think is better with what we are trying to accomplish? They are both valid approaches in BigTest.

Reply:
Hmmm.... it's a good question, and I think there are two answers. If you're the _author_ of material UI, then you want to definitely be checking about the custom ripple. However, if I'm a _user_ of material ui, this is something that I will probably never use, and so they're presence will be confusing.

I think we should optimize for the user however.

But, we shouldn't leave component authors out in the cold! Maybe we could write a custom matcher called `withCustomRipple` that would only be available in the test suite.

```ts
export const hasCustomRipple (interactor) => interactor.find(Span({ className: including('touch-ripple') })).exists();
```

Then the test becomes:
```
await hasCustomRipple(button);
```

We could even make this into a custom assertion 

```ts
export function HasTouchRipple(interactor): Assertion {
  return {
    description: `${interactor.description} has "touch ripple" effect`,
    check: () => interactor.find(Span({ className: including('touch-ripple') {))
  };
}
```

Then we can use it straight inside of an assertion:

```ts
assertion(HasTouchRipple(button))
```